server/src/main/java/org/opensearch/gateway/remote/model/RemoteGlobalMetadata.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.gateway.remote.model;
+
+import org.opensearch.cluster.metadata.Metadata;
+import org.opensearch.common.io.Streams;
+import org.opensearch.common.remote.AbstractRemoteWritableBlobEntity;
+import org.opensearch.common.remote.BlobPathParameters;
+import org.opensearch.core.compress.Compressor;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.gateway.remote.ClusterMetadataManifest.UploadedMetadata;
+import org.opensearch.repositories.blobstore.ChecksumBlobStoreFormat;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import static org.opensearch.gateway.remote.RemoteClusterStateUtils.METADATA_NAME_FORMAT;
+
+public class RemoteGlobalMetadata extends AbstractRemoteWritableBlobEntity<Metadata> {
+
+    public static final ChecksumBlobStoreFormat<Metadata> GLOBAL_METADATA_FORMAT = new ChecksumBlobStoreFormat<>(
+        "metadata",
+        METADATA_NAME_FORMAT,
+        Metadata::fromXContent
+    );
+
+    private Metadata metadata;
+    private final String blobName;
+
+    public RemoteGlobalMetadata(
+        final String blobName,
+        final String clusterUUID,
+        final Compressor compressor,
+        final NamedXContentRegistry namedXContentRegistry
+    ) {
+        super(clusterUUID, compressor, namedXContentRegistry);
+        this.blobName = blobName;
+    }
+
+    @Override
+    public BlobPathParameters getBlobPathParameters() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public String generateBlobFileName() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public UploadedMetadata getUploadedMetadata() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public InputStream serialize() throws IOException {
+        throw new UnsupportedOperationException();
+    }
```

Comment:
For my understanding why are these unimplemented

Reply:
Global metadata was being published as single file till version 2.14. From 2.15 version onwards, we are splitting it into different components. Each of these components will be responsible for serde.
So for global metadata we only need deserialization to support bwc.